erigon-lib/state/domain.go

```diff
@@ -1215,48 +1215,19 @@ func (d *Domain) buildAccessor(ctx context.Context, fromStep, toStep uint64, dat
 }
 
 func (d *Domain) missedBtreeAccessors() (l []*filesItem) {
-	d.dirtyFiles.Walk(func(items []*filesItem) bool { // don't run slow logic while iterating on btree
-		for _, item := range items {
-			fromStep, toStep := item.startTxNum/d.aggregationStep, item.endTxNum/d.aggregationStep
-			fPath := d.kvBtFilePath(fromStep, toStep)
-			exists, err := dir.FileExist(fPath)
-			if err != nil {
-				panic(err)
-			}
-			if !exists {
-				l = append(l, item)
-				continue
-			}
-			fPath = d.kvExistenceIdxFilePath(fromStep, toStep)
-			exists, err = dir.FileExist(fPath)
-			if err != nil {
-				panic(err)
-			}
-			if !exists {
-				l = append(l, item)
-				continue
-			}
+	return fileItemsWithMissingAccessors(d.dirtyFiles, d.aggregationStep, func(fromStep uint64, toStep uint64) []string {
+		return []string{
+			d.kvBtFilePath(fromStep, toStep),
```

Comment:
here need add: if d.indexList.Has(withMapIndex), if d.indexList.Has(withBtInex), if d.indexList.Has(withExistanceIndex)`.

now commitment domain has only .kvi accessor - no .bt/.kvei

Reply:
even without the checks it works because the caller "continued" the loop if index wasn't present. But look at edit now...I think it's better.